src/dali_model.h

```diff
@@ -86,23 +87,108 @@ class DaliModel : public ::triton::backend::BackendModel {
     TRITON_CALL_GUARD(
         TRITONBACKEND_ModelRepository(triton_model_, &artifact_type, &model_repo_path));
 
-    std::stringstream dali_pipeline_path;
-    dali_pipeline_path << model_repo_path << sep << version_ << sep << GetModelFilename();
-    std::string filename = dali_pipeline_path.str();
-    LOG_MESSAGE(TRITONSERVER_LOG_INFO,
-                (make_string("Loading DALI pipeline from file ", filename).c_str()));
-    dali_model_provider_ = std::make_unique<FileModelProvider>(filename);
+    std::stringstream model_path_ss;
+    model_path_ss << model_repo_path << sep << version_ << sep;
+    std::string model_path = model_path_ss.str();
+
+    std::stringstream default_model, fallback_model;
+    default_model << model_path << GetModelFilename();
+    fallback_model << model_path << fallback_model_filename_;
+
+    LoadModel(default_model.str(), fallback_model.str());
   }
 
+  /**
+   * Loads a model via the ModelProvider.
+   *
+   * @param default_model_filename The full path of the model file, specified by the user in the
+   *                               model configuration.
+   * @param fallback_model_filename The full path of the model file, which is a fallback from the
+   *                                default path, in case it is unavailable. The fallback model
+   *                                may only represent unserialized model (which will be a subject
+   *                                for autoserialization).
+   */
+  void LoadModel(std::string default_model_filename, std::string fallback_model_filename) {
+    std::string& model_filename = default_model_filename;
+    std::string target = make_string(".serialized.model.", timestamp(), ".dali");
```

Comment:
Is it fine to do this in the current working directory, or would a temporary file be more suitable?

Reply:
I would think that we are just producing garbage in the user's directory, but as you wish.